metadata/src/main/java/org/apache/kafka/image/ClusterImage.java

```diff
@@ -34,9 +34,15 @@ public final class ClusterImage {
     public static final ClusterImage EMPTY = new ClusterImage(Collections.emptyMap());
 
     private final Map<Integer, BrokerRegistration> brokers;
+    private final Map<Integer, BrokerRegistration> zkBrokers;
 
     public ClusterImage(Map<Integer, BrokerRegistration> brokers) {
         this.brokers = Collections.unmodifiableMap(brokers);
+        this.zkBrokers = Collections.unmodifiableMap(brokers
```

Comment:
Generally, we try not to store the same metadata twice in-memory. Can we just use an accessor that filters for ZK brokers rather than storing them here?

Reply:
I thought about this. Have to not strong preference since methods based on zkBrokers should not be called often. Will change.